protocol/statetracker/updaters/provider_availability_updater.go

```diff
@@ -0,0 +1,87 @@
+package updaters
+
+import (
+	"context"
+	"sync"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/lavanet/lava/v3/protocol/lavasession"
+	"github.com/lavanet/lava/v3/protocol/metrics"
+	"github.com/lavanet/lava/v3/x/pairing/types"
+	"google.golang.org/grpc"
+)
+
+type ProviderPairingStatusStateQueryInf interface {
+	Providers(ctx context.Context, in *types.QueryProvidersRequest, opts ...grpc.CallOption) (*types.QueryProvidersResponse, error)
+}
+
+type AvailabilityStatus uint64
+
+const (
+	AVAILABLE AvailabilityStatus = iota
+	FROZEN
+)
+
+type ProviderAvailabilityUpdater struct {
```

Comment:
Again we have the issue of multiple API interfaces, and chain ids. 

keep in mind. we want as least amount of calls and code being executed that is not required.

if we have the following setup:
Lava - (tendermint, rest, grpc)
Ethereum (jsonrpc)

you have created 4 updaters executing unnecessary code for LAVA - rest/grpc 

but the difference from what we did with the load manager, is that updaters already have some built-in mechanism to identify redundancy

please read carefully and make sure it is understood: 

how the raw updater works:

https://github.com/lavanet/lava/blob/7dda9b6dfe99973e53d31ff2ead881a3ba3ea662/protocol/statetracker/provider_state_tracker.go#L53
 https://github.com/lavanet/lava/blob/7dda9b6dfe99973e53d31ff2ead881a3ba3ea662/protocol/statetracker/state_tracker.go#L184
 
 and a good example for your scenario would be this updater key:
 
https://github.com/lavanet/lava/blob/7dda9b6dfe99973e53d31ff2ead881a3ba3ea662/protocol/statetracker/updaters/spec_updater.go#L50

Reply:
Fixed in [5b112c0](https://github.com/lavanet/lava/pull/1729/commits/5b112c07493613be8c622cf5c60b360cacf8fd82)